### ISSUE TITLE
How can I disabled the property paths of _.pick?

### ISSUE THREAD

**Issue by vincentngthu** [lodash/lodash#2824]
ref #2823 

Version: 4.17.1

I think the _.pick of this version has weird behavior

```javascript
_.pick({'socket.io':'^0.9.17', 'socket.io-client':'^0.9.17'}, 'socket.io')
//{ socket: { io: '^0.9.17' } }
//but I expect
//{'socket.io': '^0.9.17'}
```

How can I disabled the property paths?

**Comment by frakti** [comment]
This will work:

`_.pick({'socket.io':'^0.9.17', 'socket.io-client':'^0.9.17'}, [['socket.io']])`

**Comment by molily** [comment]
Apparently, the deep path support for _.omit/_.pick introduced in 4.17 changes how existing code behaves. If authors need to change their code to restore their original intent – opting out of the new behaviour – the feature should be shipped as a new major version. Alternatively, the existing API is kept intact and a new API for deep omit/pick is introduced – opting into the new behavior. Then it would be a minor version change. My 2 cents.

**Comment by michaelkrone** [comment]
For me it broke for the following code:
```
const picked = _.pick(
    { '1. Weihnachtsfeiertag': 'foo',  '2. Weihnachtsfeiertag': 'foo' },
    [  '1. Weihnachtsfeiertag', '2. Weihnachtsfeiertag' ]);

// picked is
{
 '1': { ' Weihnachtsfeiertag': 'foo'; },
 '2': { ' Weihnachtsfeiertag': 'foo'}
}
```

Was suprised that 'CHAR DOT SPACE CHAR' is treated as a path array. Is this intended behaviour?

**Comment by jdalton** [comment]
I've addressed this issue with #2826 so you won't have to use the an array escape.
I'll have a v4.17.2 patch release out sometime today.
